extensions/BugModal/web/bug_modal.js

```diff
@@ -1328,6 +1328,131 @@ $(function() {
+                .filter(anchor => {
+                    // TODO better way to get current bmo instance other than window.location?
+                    return (
+                        anchor.hostname === window.location.hostname &&
+                        anchor.pathname === pathname &&
+                        /^\d+$/.test(getHrefIdParam(anchor))
+                    )
+                })
+                .map(anchor => {
+                    return {
+                        id: getHrefIdParam(anchor),
+                        element: anchor
+                    }
+                })
+            )
+        };
+
+        const enhanceBugLinks = () => {
+            let bugData = {};
+            let bugLinks = findLinkElements('/show_bug.cgi');
+            let bugIds = bugLinks.map((bug) => bug['id']).filter(filterUnique).join(',');
+            if(!bugIds) return;
+
+            $.get("/rest/bug", {
+                 id: bugIds, 
+                 Bugzilla_api_token: BUGZILLA.api_token 
+            })
+            .then(responseData => {
+                responseData['bugs'].forEach(bug => bugData[bug['id']] = bug);
+            })
+            .then(() => {
+                bugLinks.forEach(bugLink => {
+                    let data = bugData[bugLink['id']];
+                    if(data) {
+                        bugLink.element.setAttribute("title", `${data.status} - ${data.summary}`);
+                        $(bugLink.element).tooltip({
+                            position: { my: "left top+8", at: "left bottom", collision: "flipfit" },
+                            show: { effect: 'none' },
+                            hide: { effect: 'none' }
+                        });
+                    }
+                });
+            })
+            .catch(e => { e; /* TODO exceptions? */ });
+        };
+
+        const enhanceAttachmentLinks = () => {
+            let attachmentData = {};
+            let attachmentLinks = findLinkElements('/attachment.cgi');
+            let attachmentIds = attachmentLinks.map(attachment => attachment['id']).filter(filterUnique);
+            if(!attachmentIds) return;
+
+            // Fetch all attachments for this bug. This one request will get data for
+            // most of the attachment links on the page.
+            $.get(`/rest/bug/${BUGZILLA.bug_id}/attachment`, { 
+                Bugzilla_api_token: BUGZILLA.api_token,
+                exclude_fields: 'data' // Remove the large Base64 encoded string of the attachment
+            })
+            .then(responseData => {
+                if(responseData['bugs'][BUGZILLA.bug_id]) {
+                    responseData['bugs'][BUGZILLA.bug_id].forEach(attachment => {
+                        attachmentData[attachment['id']] = attachment
+                    });
+                }
+            })
+            .then(() => {
+                // Individually fetch attachment data for each attachment id that wasn't found above.
```

Comment:
Found this out the hard way. This code here tries to limit the requests to bmo by querying by the current bug first (which returns all of that bug's attachments at once) which I imagine would cover most of the attachment links on a bug page. For those other attachment links, we have to query them one by one to see if the user has permission to view it :/